src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,228 @@
+    {
+        if (level == LEVEL_AT_WHICH_TO_NOT_TRIGGER_RAID)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = (int) (BARBARIANS_MULTIPLIER * level);
+        int numberOfArcherBarbarians = (int) (ARCHER_BARBARIANS_MULTIPLIER * level);
+        int numberOfChiefBarbarians = (int) (CHIEF_BARBARIANS_MULTIPLIER * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > PREFERRED_MAX_HORDE_SIZE && MAX_SIZE == PREFERRED_MAX_HORDE_SIZE)
+        {
+            //set the preferred horde style if the total spawns is greater that the config's max size
+            numberOfBarbarians = PREFERRED_MAX_BARBARIANS;
+            numberOfArcherBarbarians = PREFERRED_MAX_ARCHERS;
+            numberOfChiefBarbarians = PREFERRED_MAX_CHIEFS;
+        }
+        else if (hordeTotal > MAX_SIZE)
+        {
+            //Equilize the spawns so that there is less spawns than the config's max size
+            numberOfBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfArcherBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfArcherBarbarians);
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            numberOfChiefBarbarians = equilizeBarbarianSpawns(hordeTotal, numberOfChiefBarbarians);
+        }
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+
+        final BlockPos targetSpawnPoint = calculateSpawnLocation(raidingWorld, colony);
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, targetSpawnPoint, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, targetSpawnPoint, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, targetSpawnPoint, raidingWorld);
+    }
+
+    /**
+     * Reduces barbarian spawns to less than the maximum allowed (set via the config)
+     *
+     * @param total    The current horde size
+     * @param numberOf The number of barbarians which we are reducing
+     * @return the new number that the barbarians should be set to.
+     */
+    private static int equilizeBarbarianSpawns(final int total, final int numberOf)
+    {
+        int returnValue = numberOf;
+        if (total > Configurations.maxBarbarianHordeSize)
+        {
+            returnValue = total - MAX_SIZE;
+
+            if (returnValue < 0)
+            {
+                return 0;
+            }
+            return returnValue;
+        }
+        return returnValue;
+    }
+
+    public static boolean raidThisNight(World world)
+    {
+        float chance = 1 / Configurations.averageNumberOfNightsBetweenRaids;
+        if (world.rand.nextFloat() < chance)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)